conan/tools/env/environment.py

```diff
@@ -3,6 +3,7 @@
 import textwrap
 from collections import OrderedDict
 
+from conans.client.tools.env import _environment_add
```

Comment:
The goal is not to depend on anything of the previous ``conans.client.tools``, but to create new helpers here. The thing is that ``_environment_add()`` has all the legacy, like checking if something is a list and doing ``os.pathsep.join()`` logic directly there. 

Maybe not necessary to do as part of this PR if it works so far, but certainly a TODO